rbs/TypeParamsToParserNodes.cc

```diff
@@ -0,0 +1,75 @@
+#include "rbs/TypeParamsToParserNodes.h"
+
+#include "core/errors/rewriter.h"
+#include "parser/helper.h"
+#include "parser/parser.h"
+#include "rbs/TypeToParserNode.h"
+#include "rbs/rbs_common.h"
+
+using namespace std;
+
+namespace sorbet::rbs {
+
+parser::NodeVec TypeParamsToParserNode::typeParams(const rbs_node_list_t *rbsTypeParams,
+                                                   const RBSDeclaration &declaration) {
+    parser::NodeVec result;
+
+    for (rbs_node_list_node_t *list_node = rbsTypeParams->head; list_node != nullptr; list_node = list_node->next) {
+        ENFORCE(list_node->node->type == RBS_AST_TYPE_PARAM,
+                "Unexpected node type `{}` in type parameter list, expected `{}`", rbs_node_type_name(list_node->node),
+                "TypeParam");
+
+        auto rbsTypeParam = (rbs_ast_type_param_t *)list_node->node;
+        auto loc = declaration.typeLocFromRange(list_node->node->location->rg);
+
+        if (rbsTypeParam->unchecked) {
+            if (auto e = ctx.beginError(loc, core::errors::Rewriter::RBSUnsupported)) {
+                e.setHeader("`{}` type parameters are not supported by Sorbet", "unchecked");
+            }
+        }
+
+        auto nameStr = parser.resolveConstant(rbsTypeParam->name);
+        auto nameUTF8 = ctx.state.enterNameUTF8(nameStr);
+        auto nameConstant = ctx.state.enterNameConstant(nameUTF8);
```

Comment:
```suggestion
        auto nameConstant = ctx.state.enterNameConstant(nameStr);
```

I believe that there is an overload that does this in one line if you'd prefer